Explorer/Assets/Scripts/ECS/Unity/Materials/Systems/StartMaterialsLoadingSystem.cs

```diff
@@ -181,14 +181,26 @@ private bool TryCreateGetTexturePromise(in TextureComponent? textureComponent, r
                 FilterMode = textureComponentValue.FilterMode,
             };
 
-            promise = textureComponent.Value.IsVideoTexture
-                ? Promise.CreateFinalized(intention, GetOrAddVideoTextureResult(textureComponentValue))
-                : Promise.Create(World!, intention, partitionComponent);
+            if (textureComponent.Value.IsVideoTexture)
+                promise = Promise.CreateFinalized(intention, GetOrAddVideoTextureResult(textureComponentValue));
+            else
+            {
+                if (Equals(ref textureComponentValue, ref promise))
+                {
+                    promise = promise!.Value.TryGetResult(World!, out var texture)
```

Comment:
Nitpick: You could consider adding the `TryGetResult` as an "and" condition to the `if` in line 188, thus removing the need of ternary operator's else statement, since it's the same as the `else` of the `if` in line 188:

```csharp
if (Equals(ref textureComponentValue, ref promise) && 
   promise!.Value.TryGetResult(World!, out var texture))
{
   promise = Promise.CreateFinalized(intention, texture);
}
else
{
   promise = Promise.Create(World!, intention, partitionComponet);
}
```

Reply:
True, thx for the input!